app/src/test/java/de/westnordost/streetcomplete/quests/crossing/AddCrossingTest.kt

```diff
@@ -0,0 +1,186 @@
+package de.westnordost.streetcomplete.quests.crossing
+
+import de.westnordost.streetcomplete.quests.TestMapDataWithGeometry
+import de.westnordost.streetcomplete.testutils.node
+import de.westnordost.streetcomplete.testutils.p
+import de.westnordost.streetcomplete.testutils.way
+import org.junit.Assert.*
+import org.junit.Test
+
+class AddCrossingTest {
+    private val questType = AddCrossing()
+
+    @Test fun `free-floating nodes do not count`() {
+        val mapData = TestMapDataWithGeometry(listOf(
+            node(1)
+        ))
+        assertEquals(0, questType.getApplicableElements(mapData).toList().size)
+    }
+
+    /*
+      ══╪══
+    */
+    @Test fun `simple crossing counts`() {
+        val shared = node(2, p(0.0, 0.0))
+        val mapData = TestMapDataWithGeometry(listOf(
+            node(1, p(0.0, -1.0)),
+            shared,
+            node(3, p(0.0, +1.0)),
+            node(4, p(-1.0, 0.0)),
+            node(5, p(+1.0, 0.0)),
+            way(1, nodes = listOf(1, 2, 3), tags = mapOf("highway" to "unclassified")),
+            way(2, nodes = listOf(4, 2, 5), tags = mapOf("highway" to "footway")),
+        ))
+        assertEquals(shared, questType.getApplicableElements(mapData).toList().single())
+    }
+
+    /*
+      ═══╡
+    */
+    @Test fun `crossing road on end node does not count`() {
+        val shared = node(2, p(0.0, 0.0))
+        val mapData = TestMapDataWithGeometry(listOf(
+            node(1, p(0.0, -1.0)),
+            shared,
+            node(4, p(-1.0, 0.0)),
+            node(5, p(+1.0, 0.0)),
+            way(1, nodes = listOf(1, 2), tags = mapOf("highway" to "unclassified")),
+            way(2, nodes = listOf(4, 2, 5), tags = mapOf("highway" to "footway")),
+        ))
+        assertEquals(0, questType.getApplicableElements(mapData).toList().size)
+    }
+
+    /*
+      ══╧══
+    */
+    @Test fun `crossing footway on end node does not count`() {
+        val shared = node(2, p(0.0, 0.0))
+        val mapData = TestMapDataWithGeometry(listOf(
+            node(1, p(0.0, -1.0)),
+            shared,
+            node(3, p(0.0, +1.0)),
+            node(4, p(-1.0, 0.0)),
+            way(1, nodes = listOf(1, 2, 3), tags = mapOf("highway" to "unclassified")),
+            way(2, nodes = listOf(4, 2), tags = mapOf("highway" to "footway")),
+        ))
+        assertEquals(0, questType.getApplicableElements(mapData).toList().size)
+    }
```

Comment:
This is what it looks like at the boundary between separated sidewalks and shared-way sidewalks, and it represents a crossing at that boundary: https://wiki.openstreetmap.org/wiki/File:Sidewalk_crossing-EN.png. It probably makes sense to present this quest when the `highway=unclassified` here also has a `sidewalk=*` key indicating there's a sidewalk on the opposite side of the road from the separated footway.

Reply:
To paraphrase your suggestion...:

Treat a footway segment as **crossing** a road even if it only touches it, **if** the road has a sidewalk on the other side of the road (as seen from that footway segment).

This could work.

However, at least for the first iteration/implementation, I think I would rather be more conservative here because maybe I am missing a possible configuration. After all, we kind of need to keep in mind also kind of wrong-ish taggings that don't follow the ideal tagging strictly.

The crossover between the two tagging schemes are always problematic. In general, we need to be wary to not produce false positives, i.e. the quest definitely should not show if a footway just ends on a road (or touches it) but does not cross it.

So also because it increases the complexity of the implementation, I would not implement that for the time being. After all, if any `highway=crossing` is tagged, we already ask for the properties of a crossing. This is just to find the places where this info is missing with relative certainty.